app/tests/fixtures/credentials.py

```diff
@@ -433,3 +433,80 @@ async def get_or_issue_regression_cred_valid(
         referent=valid_credential["referent"],
         cred_def_revocable=valid_credential["cred_def_id"],
     )
+
+
+@pytest.fixture(scope="function")
+async def issue_alice_many_creds(
+    request,
+    faber_client: RichAsyncClient,
+    alice_member_client: RichAsyncClient,
+    credential_definition_id: str,
+    faber_and_alice_connection: FaberAliceConnect,
+) -> List[CredentialExchange]:
+
+    faber_conn_id = faber_and_alice_connection.faber_connection_id
+
+    faber_cred_ex_ids = []
+    for i in range(request.param):
+        credential = {
+            "connection_id": faber_conn_id,
+            "save_exchange_record": True,
+            "indy_credential_detail": {
+                "credential_definition_id": credential_definition_id,
+                "attributes": {"speed": str(i), "name": "Alice", "age": "44"},
+            },
+        }
+        response = (
+            await faber_client.post(
+                CREDENTIALS_BASE_PATH,
+                json=credential,
+            )
+        ).json()
+
+        faber_cred_ex_id = response["credential_exchange_id"]
+        faber_cred_ex_ids += [faber_cred_ex_id]
+
+        thread_id = response["thread_id"]
+
+        await check_webhook_state(
+            client=alice_member_client,
+            topic="credentials",
+            state="offer-received",
+            filter_map={
+                "thread_id": thread_id,
+            },
+        )
+
+        cred_exchange = (
+            await alice_member_client.get(
+                f"{CREDENTIALS_BASE_PATH}?thread_id={thread_id}"
+            )
+        ).json()[0]
```

Comment:
In general: optimise for readability. This is awaiting, calling .json(), and taking first element of list. Rather split it up:

```py
response = await alice_member_client.get(
                f"{CREDENTIALS_BASE_PATH}?thread_id={thread_id}"
            )
cred_exchange = response.json()[0]
```

And later:
```py
        await alice_member_client.post(
            f"{CREDENTIALS_BASE_PATH}/{cred_exchange['credential_exchange_id']}/request",
            json={},
        )
```
^ rather assign `cred_exchange['credential_exchange_id']` to a variable and use that. Again for readability - please always prefer doing it this way.
___

But here's a better way. Capture the webhook that was awaited by thread_id, and get alice's cred_ex_id from there:
```py
        alice_event = await check_webhook_state(
            client=alice_member_client,
            topic="credentials",
            state="offer-received",
            filter_map={
                "thread_id": thread_id,
            },
        )
        alice_cred_ex_id = alice_event ["credential_exchange_id"]
```

and then there's no need to fetch the record and do `cred_exchange['credential_exchange_id']` - can just directly do the `POST /request`